app/actions/remote/post.ts

```diff
@@ -552,3 +552,45 @@ export const togglePinPost = async (serverUrl: string, postId: string) => {
         return {error};
     }
 };
+
+export const markPostAsUnread = async (serverUrl: string, postId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const [userId, post] = await Promise.all([queryCurrentUserId(database), queryPostById(database, postId)]);
```

Comment:
will the queries be changing with #5984 ? interesting use case
cc: @larkox

Reply:
This would change as to `Promise.all([getCurrentUserId(database), getPost(database, postId)]);`. So... not a big deal of a change.